src/Ordering.API/Application/Queries/OrderQueries.cs

```diff
@@ -1,85 +1,57 @@
-            new { id });
-
-        if (result.AsList().Count == 0)
+        var order = await orderRepository.GetAsync(id);
+        if (order is null)
             throw new KeyNotFoundException();
 
-        return MapOrderItems(result);
-    }
-
-    public async Task<IEnumerable<OrderSummary>> GetOrdersFromUserAsync(string userId)
-    {
-        using var connection = dataSource.OpenConnection();
-
-        return await connection.QueryAsync<OrderSummary>("""
-            SELECT o."Id" AS ordernumber, o."OrderDate" AS date, os."Name" AS status, SUM(oi."Units" * oi."UnitPrice") AS total
-            FROM ordering.orders AS o
-            LEFT JOIN ordering."orderItems" AS oi ON o."Id" = oi."OrderId"
-            LEFT JOIN ordering.orderstatus AS os ON o."OrderStatusId" = os."Id"
-            LEFT JOIN ordering.buyers AS ob ON o."BuyerId" = ob."Id"
-            WHERE ob."IdentityGuid" = @userId
-            GROUP BY o."Id", o."OrderDate", os."Name"
-            ORDER BY o."Id"
-            """,
-            new { userId });
-    }
-
-    public async Task<IEnumerable<CardType>> GetCardTypesAsync()
-    {
-        using var connection = dataSource.OpenConnection();
+        return new Order
+        {
+            ordernumber = order.Id,
+            date = order.GetOrderDate(),
+            description = order.GetDescription(),
+            city = order.Address.City,
+            country = order.Address.Country,
+            state = order.Address.State,
+            street = order.Address.Street,
+            zipcode = order.Address.ZipCode,
+            status = order.OrderStatus.Name,
+            total = order.GetTotal(),
+            orderitems = order.OrderItems.Select(oi => new Orderitem
+            {
+                productname = oi.GetOrderItemProductName(),
+                units = oi.GetUnits(),
+                unitprice = (double)oi.GetUnitPrice(),
+                pictureurl = oi.GetPictureUri()
+            }).ToList()
+        };
 
-        return await connection.QueryAsync<CardType>("SELECT * FROM ordering.cardtypes");
     }
 
-    private Order MapOrderItems(dynamic result)
+    public async Task<IEnumerable<OrderSummary>> GetOrdersFromUserAsync(string userId)
     {
-        var order = new Order
-        {
-            ordernumber = result[0].ordernumber,
-            date = result[0].date,
-            status = result[0].status,
-            description = result[0].description,
-            street = result[0].street,
-            city = result[0].city,
-            state = result[0].state,
-            zipcode = result[0].zipcode,
-            country = result[0].country,
-            orderitems = new List<Orderitem>(),
-            total = 0
-        };
-
-        foreach (dynamic item in result)
-        {
-            var orderitem = new Orderitem
+        return await context.Orders
+            .Include(o => o.OrderItems)
+            .Include(o => o.OrderStatus)
+            .Join(context.Buyers,
```

Comment:
Given that Order has a BuyerId (foreign key), there should be a regular relationship between Orders and Buyers, and this should be a simple Include... I'm really not sure why it wasn't modelled this way.

But this seems OK as a temporary thing.

Reply:
I've added a navigation property to the order entity and updated the EF configuration. However, I'm hesitant about this approach due to potential conflicts with Domain-Driven Design (DDD) principles, specifically regarding aggregates. Key concerns include:

1. **Violation of Aggregate Boundaries**: Navigation properties can create tight coupling between aggregates, undermining the autonomy and clear boundaries essential in DDD.

2. **Encouragement of Lazy Loading**: The use of navigation properties often leads to lazy loading, which can result in performance issues.